consensus/polybft/contractsapi/contractsapi.go

```diff
@@ -0,0 +1,240 @@
+	"github.com/umbracle/ethgo/abi"
+)
+
+type StateSyncCommitment struct {
+	StartID *big.Int   `abi:"startId"`
+	EndID   *big.Int   `abi:"endId"`
+	Root    types.Hash `abi:"root"`
+}
+
+var StateSyncCommitmentABIType = abi.MustNewType("tuple(uint256 startId,uint256 endId,bytes32 root)")
+
+func (S *StateSyncCommitment) EncodeAbi() ([]byte, error) {
+	return StateSyncCommitmentABIType.Encode(S)
+}
+
+func (S *StateSyncCommitment) DecodeAbi(buf []byte) error {
+	return decodeStruct(StateSyncCommitmentABIType, buf, &S)
+}
+
+type Commit struct {
+	Commitment *StateSyncCommitment `abi:"commitment"`
+	Signature  []byte               `abi:"signature"`
+	Bitmap     []byte               `abi:"bitmap"`
+}
+
+func (c *Commit) EncodeAbi() ([]byte, error) {
+	return StateReceiver.Abi.Methods["commit"].Encode(c)
+}
+
+func (c *Commit) DecodeAbi(buf []byte) error {
+	return decodeMethod(StateReceiver.Abi.Methods["commit"], buf, c)
+}
+
+type StateSync struct {
+	ID       *big.Int      `abi:"id"`
+	Sender   types.Address `abi:"sender"`
+	Receiver types.Address `abi:"receiver"`
+	Data     []byte        `abi:"data"`
+}
+
+var StateSyncABIType = abi.MustNewType("tuple(uint256 id,address sender,address receiver,bytes data)")
+
+func (S *StateSync) EncodeAbi() ([]byte, error) {
+	return StateSyncABIType.Encode(S)
+}
+
+func (S *StateSync) DecodeAbi(buf []byte) error {
+	return decodeStruct(StateSyncABIType, buf, &S)
+}
+
+type Execute struct {
+	Proof []types.Hash `abi:"proof"`
+	Obj   *StateSync   `abi:"obj"`
+}
+
+func (e *Execute) EncodeAbi() ([]byte, error) {
+	return StateReceiver.Abi.Methods["execute"].Encode(e)
+}
+
+func (e *Execute) DecodeAbi(buf []byte) error {
+	return decodeMethod(StateReceiver.Abi.Methods["execute"], buf, e)
+}
+
+type StateSyncResultEvent struct {
+	Counter *big.Int `abi:"counter"`
+	Status  bool     `abi:"status"`
+	Message []byte   `abi:"message"`
+}
+
+func (s *StateSyncResultEvent) ParseLog(log *ethgo.Log) error {
+	return decodeEvent(StateReceiver.Abi.Events["StateSyncResult"], log, s)
+}
+
+type NewCommitmentEvent struct {
+	StartID *big.Int   `abi:"startId"`
+	EndID   *big.Int   `abi:"endId"`
+	Root    types.Hash `abi:"root"`
+}
+
+func (n *NewCommitmentEvent) ParseLog(log *ethgo.Log) error {
+	return decodeEvent(StateReceiver.Abi.Events["NewCommitment"], log, n)
+}
+
+type Epoch struct {
+	StartBlock *big.Int   `abi:"startBlock"`
+	EndBlock   *big.Int   `abi:"endBlock"`
+	EpochRoot  types.Hash `abi:"epochRoot"`
+}
+
+var EpochABIType = abi.MustNewType("tuple(uint256 startBlock,uint256 endBlock,bytes32 epochRoot)")
+
+func (E *Epoch) EncodeAbi() ([]byte, error) {
+	return EpochABIType.Encode(E)
+}
+
+func (E *Epoch) DecodeAbi(buf []byte) error {
+	return decodeStruct(EpochABIType, buf, &E)
+}
+
+type UptimeData struct {
+	Validator    types.Address `abi:"validator"`
+	SignedBlocks *big.Int      `abi:"signedBlocks"`
+}
+
+var UptimeDataABIType = abi.MustNewType("tuple(address validator,uint256 signedBlocks)")
```

Comment:
Does this needs to be public?

Reply:
We need these public, because we use some of them to create hashes for certain things. For example, commitment hash.